apps/cli/src/commands/doctor.ts

```diff
@@ -75,9 +83,21 @@ export default class DoctorCommand extends Command {
             this.log("Your system is ready for sunodo.");
         } catch (error: unknown) {
             if (error instanceof Error) {
-                this.error(error.message);
-            } else {
-                this.error(String(error));
+                if (error.message.includes("ENOENT")) {
+                    if (error.message.includes("docker --version")) {
+                        this.error(
+                            "Docker is required but not installed or the command execution failed. Please refer to the Docker documentation for installation instructions: https://docs.docker.com/get-docker/",
+                        );
+                    } else if (error.message.includes("docker compose")) {
+                        this.error(
+                            "Docker Compose is required but not installed or the command execution failed. Please refer to the Docker Compose documentation for installation instructions: https://docs.docker.com/compose/install/",
+                        );
+                    } else {
+                        this.error(error.message);
+                    }
+                } else {
+                    this.error(String(error));
+                }
```

Comment:
You're relying on error messages when you should rely on exit codes.

I suggest you have individual `try...catch` blocks for each command being checked.

Reply:
> You're relying on error messages when you should rely on exit codes.

Property 'code' does not exist on type 'Error'.ts(2339)

Reply:
> I suggest you have individual `try...catch` blocks for each command being checked.

Sure